File: scala/scalafind/src/test/scala/scalafind/SortByTest.scala
package scalafind

import org.scalatest.funsuite.AnyFunSuite

class SortByTest extends AnyFunSuite {
  /***************************************************************************
    * fromName tests
    **************************************************************************/
  test("test fromName FILENAME") {
    assert(SortBy.forName("FILENAME").equals(SortBy.FileName))
  }

  test("test fromName filename") {
    assert(SortBy.forName("filename").equals(SortBy.FileName))
  }

  test("test fromName name") {
    assert(SortBy.forName("name").equals(SortBy.FileName))
  }

  test("test fromName FILESIZE") {
    assert(SortBy.forName("FILESIZE").equals(SortBy.FileSize))
  }

  test("test fromName filesize") {
    assert(SortBy.forName("filesize").equals(SortBy.FileSize))
  }

  test("test fromName size") {
    assert(SortBy.forName("size").equals(SortBy.FileSize))
  }

  test("test fromName FILETYPE") {
    assert(SortBy.forName("FILETYPE").equals(SortBy.FileType))
  }

  test("test fromName filetype") {
    assert(SortBy.forName("filetype").equals(SortBy.FileType))
  }

  test("test fromName type") {
    assert(SortBy.forName("type").equals(SortBy.FileType))
  }

  test("test fromName LASTMOD") {
    assert(SortBy.forName("LASTMOD").equals(SortBy.LastMod))
  }

  test("test fromName lastmod") {
    assert(SortBy.forName("lastmod").equals(SortBy.LastMod))
  }

  test("test fromName FILEPATH") {
    assert(SortBy.forName("FILEPATH").equals(SortBy.FilePath))
  }

  test("test fromName filepath") {
    assert(SortBy.forName("filepath").equals(SortBy.FilePath))
  }

  test("test fromName anything") {
    assert(SortBy.forName("anything").equals(SortBy.FilePath))
  }
}

File: kotlin/ktfind/lib/src/test/kotlin/ktfind/SortByTest.kt
package ktfind

import kotlin.test.Test
import kotlin.test.assertEquals

/**
 * @author cary on 7/30/16.
 */
class SortByTest {

    /***************************************************************************
     * fromName tests
     **************************************************************************/
    @Test
    fun testFromNameFileNameAllUppercase() {
        val sortByName = "FILENAME"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILENAME, sortBy)
    }

    @Test
    fun testFromNameFileNameAllLowercase() {
        val sortByName = "filename"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILENAME, sortBy)
    }

    @Test
    fun testFromNameName() {
        val sortByName = "NAME"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILENAME, sortBy)
    }

    @Test
    fun testFromNameFileSizeAllUppercase() {
        val sortByName = "FILESIZE"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILESIZE, sortBy)
    }

    @Test
    fun testFromNameFileSizeAllLowercase() {
        val sortByName = "filesize"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILESIZE, sortBy)
    }

    @Test
    fun testFromNameSize() {
        val sortByName = "size"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILESIZE, sortBy)
    }

    @Test
    fun testFromNameFileTypeAllUppercase() {
        val sortByName = "FILETYPE"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILETYPE, sortBy)
    }

    @Test
    fun testFromNameFileTypeAllLowercase() {
        val sortByName = "filetype"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILETYPE, sortBy)
    }

    @Test
    fun testFromNameType() {
        val sortByName = "type"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILETYPE, sortBy)
    }

    @Test
    fun testFromNameLastModAllUppercase() {
        val sortByName = "LASTMOD"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.LASTMOD, sortBy)
    }

    @Test
    fun testFromNameLastModAllLowercase() {
        val sortByName = "lastmod"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.LASTMOD, sortBy)
    }

    @Test
    fun testFromNameFilePathAllUppercase() {
        val sortByName = "FILEPATH"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILEPATH, sortBy)
    }

    @Test
    fun testFromNameFilePathAllLowercase() {
        val sortByName = "filepath"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILEPATH, sortBy)
    }

    @Test
    fun testFromNameAnything() {
        val sortByName = "anything"
        val sortBy = SortBy.forName(sortByName)
        assertEquals(SortBy.FILEPATH, sortBy)
    }
}
